templates/junit4.java.qute

```diff
@@ -0,0 +1,54 @@
+///usr/bin/env jbang "$0" "$@" ; exit $?
+
+//DEPS junit:junit-dep:4.11
+
+//SOURCES {scriptName}.java
+
+import org.junit.Test;
+import org.junit.runner.JUnitCore;
+import org.junit.runner.Request;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+
+import static java.lang.System.out;
+
+// JUnit4 Test class for {scriptName}
+public class {baseName} {
+
+    // Run all Unit tests with JBang with ./{baseName}.java
+    public static void main(final String... args) {
+        final Request request = Request.aClass({baseName}.class);
+        final JUnitCore runner = new JUnitCore();
+        final Result result = runner.run(request);
+        out.println(summary(result));
+    }
+
+    // Define each Unit test here and run them separately in the IDE
+    @Test
+    public void test() {
+    }
+
+    private static String summary(final Result result) {
```

Comment:
why is this custom printing needed ? 

would it not be sufficient with:

```
       final SummaryGeneratingListener generatingListener = new SummaryGeneratingListener();
        launcher.execute(request, generatingListener);
        generatingListener.getSummary().printFailuresTo(new java.io.PrintWriter(out));
        generatingListener.getSummary().printTo(new java.io.PrintWriter(out));
```
i.e. use the printing methods of SummaryGeneratingListener ?

Reply:
We can do that to simplify but only for JUnit 5 template: the JUnit platform library providing `SummaryGeneratingListener` is JUnit 5 specific. It uses a `Launcher` while JUnit 4 uses a `Runner`.

So i use custom code for both templates to have similar output with both templates.